### PR TITLE
chore(deps): fix  CVE-2024-21536

### PR DESCRIPTION
**Pull request by kim-tsao**
## Description

Please explain the changes you made here.
This fix supersedes https://github.com/janus-idp/backstage-showcase/pull/1809
Bump up `http-proxy-middleware` to v2.0.7 and v3.0.3 to fix CVE-2024-21536

## Which issue(s) does this PR fix

- Fixes #?
- https://issues.redhat.com/browse/RHIDP-4612

## PR acceptance criteria

Please make sure that the following steps are complete:

- [ ] GitHub Actions are completed and successful
- [ ] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [ ] Documentation is updated if necessary (requirement for new features)
- [ ] Add a screenshot if the change is UX/UI related

### CODE REVIEW

**Comment by christoph-jerolimov** [comment]
The image was broken, @ciiay fixed an issue with #1847, let see if this helps

/retest

File: yarn.lock [hunk 1]
```diff
@@ -17561,7 +17561,16 @@ __metadata:
...
+  resolution: "@types/http-proxy@npm:1.17.15"
+  dependencies:
+    "@types/node": "*"
+  checksum: d96eaf4e22232b587b46256b89c20525c453216684481015cf50fb385b0b319b883749ccb77dee9af57d107e8440cdacd56f4234f65176d317e9777077ff5bf3
+  languageName: node
+  linkType: hard
+
+"@types/http-proxy@npm:^1.17.8":
   version: 1.17.14
   resolution: "@types/http-proxy@npm:1.17.14"
```
Comment (nickboldt) [inline yarn.lock:17575]: why do we want BOTH 1.17.14 and 1.17.15?

File: yarn.lock [hunk 2]
```diff
@@ -17603,7 +17603,16 @@ __metadata:
   languageName: node
   linkType: hard
 
-"@types/http-proxy@npm:^1.17.10, @types/http-proxy@npm:^1.17.8":
+"@types/http-proxy@npm:^1.17.15":
```
Comment (nickboldt) [inline yarn.lock:17606]:
```suggestion
"@types/http-proxy@npm:^1.17.10, @types/http-proxy@npm:^1.17.8", "@types/http-proxy@npm:^1.17.15":
```

File: yarn.lock [hunk 3]
```diff
@@ -17603,7 +17603,16 @@ __metadata:
...
+  version: 1.17.15
+  resolution: "@types/http-proxy@npm:1.17.15"
+  dependencies:
+    "@types/node": "*"
+  checksum: d96eaf4e22232b587b46256b89c20525c453216684481015cf50fb385b0b319b883749ccb77dee9af57d107e8440cdacd56f4234f65176d317e9777077ff5bf3
+  languageName: node
+  linkType: hard
+
+"@types/http-proxy@npm:^1.17.8":
```
Comment (nickboldt) [inline yarn.lock:17615]: can delete this block if we resolve 1.17.8 as 1.17.15, above.

**Review by nickboldt — APPROVED** [review]

**Pull request merged**